stf.test/build.gradle

```diff
@@ -21,37 +25,58 @@ dependencies {
     // TODO: The custom saros eclipse plugin adds all dependencies to implementation
     stfTestCompile configurations.implementation
     testCompile configurations.stfTestCompile // for running the tests in eclipse
-}
 
-// TODO move into buildSrc
-task stfTest (type: Test, dependsOn: [':saros.eclipse:build']) {
-    group 'Verification'
-    description 'Runs the stf tests. Requires a corresponding test environment'
-
-    testClassesDirs = sourceSets.stfTest.output.classesDirs
+    stfFlakyAndFailingTestCompile project(':saros.stf')
+    stfFlakyAndFailingTestCompile project(path: ':saros.stf', configuration: 'testing')
+    stfFlakyAndFailingTestCompile configurations.testConfig
+    stfFlakyAndFailingTestCompile configurations.implementation
+    testCompile configurations.stfFlakyAndFailingTestCompile // for running the tests in eclipse
+}
 
+class StfTest extends Test {
```

Comment:
Ok, this confuses me. This looks like java syntax in the context of a gradle build file. Could you give some more detail on what is happening here?

Reply:
A gradle build file uses a Groovy DSL syntax which allows using common Groovy code (which has a syntax that allows writing java syntax). It would be possible to convert it into Java and move it into the buildSrc dir, but I created this as an interim solution to fix all flaky/failing stf tests and want to remove the logic after we fixed the tests.

Reply:
> Could you give some more detail on what is happening here?

Both stf tasks share multiple configuration. I created this task class in order to avoid duplication.